module1/homework/grayscaleImages/compression.hpp

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <array>
+#include <vector>
+
+constexpr size_t width = 32;
+constexpr size_t height = 32;
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& arr);
+std::array<std::array<uint8_t, width>, height> decompressGrayscale(const std::vector<std::pair<uint8_t, uint8_t>>& vec);
```

Comment:
```suggestion
using Image = t std::array<std::array<uint8_t, width>, height>;
using compressedImage =  std::vector<std::pair<uint8_t, uint8_t>>;
compressedImage compressGrayscale(const Image& bitmap);
Image decompressGrayscale(const compressedImage& compression);
```

Reply:
You should consider make aliases for longer types like Vicx95 suggested.